### PR TITLE
Copy - 3393 - Update profile back button text

### LINKED ISSUE #3393: Buttons on some profile pages without actions should have different copy.
## Description
On most of the Profile sub-pages, there are form elements to complete on the page and you can either save changes at the bottom of page or "Cancel and go back" at the top of the page.

For the "Diversity and Inclusion" sub-page and the "My skills and experience" sub-page however, you actually save inside a modal or on another page. In this context, the "Cancel and go back" action doesn't make sense.

For these 2 pages, both buttons should simply say "Go Back to My Profile".

## Screenshot: Prototype

image

image

### PR DESCRIPTION
**Pull request by esizer**
Resolves #3393 

## Summary

This updates the props for the `ProfileFormWrapper` and `CancelButton` to components to allow more customisation. As well, it changes the cancel link text on `/talent/profile/skills-and-experiences` and `/talent/profile/diversity-and-inclusion` for improved context.

## Screenshot
Screen Shot 2022-08-11 at 9 49 09 AM

## Testing

1. Build talentsearch `cd frontend && npm run production --workspace=talentsearch`
2. Navigate to `/talent/profile/skills-and-experiences` and `/talent/profile/diversity-and-inclusion`
3. Confirm the cancel links contain the text "Go back to my profile"

### CODE REVIEW

File: frontend/talentsearch/src/js/components/applicantProfile/ExperienceAndSkills.tsx [hunk 1]
```diff
@@ -139,6 +139,12 @@ export const ExperienceAndSkills: React.FunctionComponent<
         description:
           "Heading for experience and skills page in applicant profile.",
       })}
+      cancelLink={{
+        children: intl.formatMessage({
+          defaultMessage: "Go back to my profile",
+          description: "Link text for button to return to user profile",
+        }),
+      }}
```
Comment (yonikid15) [inline frontend/talentsearch/src/js/components/applicantProfile/ExperienceAndSkills.tsx:147]: You're going to need to update the `<ProfileFormFooter />` [on line 210](https://github.com/GCTC-NTGC/gc-digital-talent/blob/dd9a2915d67008f44763b236cd68f9cff807fdd4/frontend/talentsearch/src/js/components/applicantProfile/ExperienceAndSkills.tsx#L210) to accept a cancelLink as well.

**Review by yonikid15 — CHANGES_REQUESTED** [review]
Nice work Eric, just found a couple small bugs that should take no time:

- Experience and Skills page: button on footer is incorrect.
experience_and_skills_3652

- Diversity and inclusion: Footer button is missing completely

diversity_and_inclusion_3652

File: frontend/talentsearch/src/js/components/applicantProfile/ProfileFormWrapper.tsx [hunk 2]
```diff
@@ -4,14 +4,14 @@ import { UserIcon } from "@heroicons/react/solid";
 import { useIntl } from "react-intl";
 import { imageUrl } from "@common/helpers/router";
 import TALENTSEARCH_APP_DIR from "../../talentSearchConstants";
-import CancelButton from "./CancelButton";
+import CancelButton, { type CancelButtonProps } from "./CancelButton";
```
[the remark this answers is not in the document]
Comment (esizer) [reply frontend/talentsearch/src/js/components/applicantProfile/ProfileFormWrapper.tsx:7]: I don't think it is **_necessary_**, but with some bundlers it hints to not include it as part of the bundle so I try to use it when importing something only as a type.

File: frontend/talentsearch/src/js/components/diversityEquityInclusion/DiversityEquityInclusionForm.tsx [hunk 3]
```diff
@@ -56,6 +56,12 @@ export const DiversityEquityInclusionForm: React.FC<
           }),
         },
       ]}
+      cancelLink={{
+        children: intl.formatMessage({
+          defaultMessage: "Go back to my profile",
+          description: "Link text for button to return to user profile",
+        }),
+      }}
```
[the remark this answers is not in the document]
Comment (esizer) [reply frontend/talentsearch/src/js/components/diversityEquityInclusion/DiversityEquityInclusionForm.tsx]: Thanks! I didn't realize how many different places we needed to use this, I pulled it out in [move back to profile to common messages](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3652/commits/c15c83f799b05264428f486dfc870e8cdeab5d06) and added the missing button in [add footer cancel button to dei page](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3652/commits/46050505cc7d303963b0a11f2d9e0b252ff06905)

**Review by yonikid15 — APPROVED** [review]
Looks great!

**Pull request merged**